go/vt/topo/decode.go

```diff
@@ -53,6 +53,11 @@ func DecodeContent(filename string, data []byte, json bool) (string, error) {
 		p = new(topodatapb.SrvKeyspace)
 	case RoutingRulesFile:
 		p = new(vschemapb.RoutingRules)
+	case CommonRoutingRulesFile:
+		switch path.Base(dir) {
+		case "keyspace":
+			p = new(vschemapb.KeyspaceRoutingRules)
+		}
```

Comment:
This was needed as a follow-up to: https://github.com/vitessio/vitess/pull/15807

This PR is also what allows you to easily revert to the previous value as we had toyed with the idea of returning the *old and new* values from the `ApplyKeyspaceRoutingRules` command for this purpose.